TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/ResponderTask.java

```diff
@@ -0,0 +1,93 @@
+package com.testingtech.playits.canfilter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.nio.charset.Charset;
+import java.util.TimerTask;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class ResponderTask extends TimerTask {
+
+  private String key;
+  private Car2XEntry car2xEntry;
+  private Socket socket;
+
+//  /**
+//   * Periodically sends a key-value pair encoded as a JSON String over a
+//   * socket's output stream.
+//   * 
+//   * @param key
+//   *            openXC key
+//   * @param car2xEntry
+//   *            corresponding value entry
+//   * @param socket
+//   *            TCP response channel
+//   */
+  public ResponderTask(String key, Car2XEntry car2xEntry, Socket socket) {
+    this.key = key;
+    this.car2xEntry = car2xEntry;
+    this.socket = socket;
+  }
+
+  @Override
+  public void run() {
+    try {
+      JSONObject response = createResponse();
+      System.out.println("[ResponderTask] Sending " + response);
+      sendJSONObject(response);
+    } catch (JSONException e) {
+      System.out.println("[ResponderTask] Error creating JSON response. "
+          + e.getMessage());
+    }
+  }
+
+  private JSONObject createResponse() throws JSONException {
+    // INFO: omits key if value is empty
+    JSONObject response = new JSONObject();
+    response.put("OpenXCKey", key);
+    response.put("OBD2Key", car2xEntry.getOBD2key());
+    response.put("car2XValue", getValue());
+    response.put("eventValue", car2xEntry.getEvent());
+    response.put("respTimestamp", car2xEntry.getTimestamp());
+    return response;
+  }
+
+  private JSONObject getValue() throws JSONException {
+    JSONObject valueObject = new JSONObject();
+    Object value = car2xEntry.getValue();
+    if (value instanceof String) {
+      valueObject.put("stringValue", value.toString());
+    } else if (value instanceof Double) {
+      valueObject.put("floatValue", new Float((double) value));
+    } else if (value instanceof Integer) {
+      valueObject.put("floatValue", new Float((int) value));
+    } else if (value instanceof Boolean) {
+      valueObject.put("booleanValue", (boolean) value);
+    } else {
+      valueObject.put("stringValue", "null");
+    }
+    return valueObject;
+  }
+
+  private void sendJSONObject(JSONObject jsonObject) {
+    String jsonAsString = jsonObject.toString();

```

Comment:
The conversion from JSONObject to a byte-array are not task of this method, so it should be excluded to another method. In addition it does not seem to be an effective solution for this problem. Is that really the best way?